data-service/src/main/java/uk/gov/gchq/palisade/service/data/model/AuthorisedDataRequest.java

```diff
@@ -0,0 +1,179 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.palisade.service.data.model;
+
+import uk.gov.gchq.palisade.Context;
+import uk.gov.gchq.palisade.Generated;
+import uk.gov.gchq.palisade.User;
+import uk.gov.gchq.palisade.resource.LeafResource;
+import uk.gov.gchq.palisade.rule.Rules;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.StringJoiner;
+
+/**
+ * The AuthorisedDataRequest is the reference for the data that has been requested and prepared for the client.
+ * This will then be used to provide the filtered data based on the rules in place and the context of the request.
+ */
+public final class AuthorisedDataRequest {
+
+    private final LeafResource resource; //resource that is to be viewed
+    private final User user; //user that is requesting the information
+    private final Context context;  //context of the request
+    private final Rules<?> rules;  //rules that apply to the resource
+
+    private AuthorisedDataRequest(
+            final LeafResource resource,
+            final User user,
+            final Context context,
+            final Rules<?> rules) {
+
+        this.resource = Optional.ofNullable(resource)
+                .orElseThrow(() -> new IllegalArgumentException("resource cannot be null"));
+        this.user = Optional.ofNullable(user)
+                .orElseThrow(() -> new IllegalArgumentException("user cannot be null"));
+        this.context = Optional.ofNullable(context)
+                .orElseThrow(() -> new IllegalArgumentException("context cannot be null"));
+        this.rules = Optional.ofNullable(rules)
+                .orElseThrow(() -> new IllegalArgumentException("rules cannot be null"));
+    }
+
+    @Generated
+    public LeafResource getResource() {
+        return resource;
+    }
+
+    @Generated
+    public User getUser() {
+        return user;
+    }
+
+    @Generated
+    public Context getContext() {
+        return context;
+    }
+
+    @Generated
+    public Rules<?> getRules() {
+        return rules;
+    }
+
+    /**
+     * Builder class for the creation of instances of the AuthorisedDataRequest.
+     * This is a variant of the Fluent Builder which will use Java Objects for the components
+     * in the build.
+     */
+    public static class Builder {
+        /**
+         * Starter method for the Builder class.
+         * This method is called to start the process of creating the AuthorisedDataRequest class.
+         *
+         * @return interface {@link IResource} for the next step in the build.
+         */
+        public static IResource create() {
+            return resource -> user -> context -> rules ->
```

Comment:
again, general template is that user is first, then resource

Reply:
See comment for AuditErrorMessage.